### ISSUE TITLE
A python function to pull data from Snowflake

### ISSUE THREAD

**Issue by rai-shashank** [tusharchou/local-data-platform#75]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by fuzzyjeet** [comment]
Hi,
Here's a summary of a problem I faced last week at my workplace:
tl;dr: Suddenly deleted Google groups in ACL files led to undetected run failures and a delay in gaining access

Long version:
While working on a time-sensitive ticket to add a new brand's data to a central data table, we had to request access for our Airflow service account (SA) to that brand's data domain unit (DDU). We followed the process and asked the data engineers from that brand to add our SA to their access control list (ACL). However, even long after they merged the PR granting our SA access, we could not access their tables.

Upon escalation, a central data engineer discovered that the ACL DAG in the brand's pipeline was failing for many days now. Their ACLs had not been refreshed for the past week. They discovered the root cause to be a Google group in the ACL that was inactive and was causing the error. Unfortunately, this brand and its team had newly migrated to our central data infrastructure and had not fully completed their setup - meaning, they had not set up Slack or any alerts for failures in their scheduled tasks (on Airflow). To add to this, no one had opened the ACL DAG to check on its status in a while. All this led to an undetected task failure that ended up impacting a time-critical task at the n-th moment.

We further identified that an unplanned/uninformed cleanup of Google groups by another team had led to a bunch of requests to the data engineers to purge some group IDs from the Central ACLs, but these were not informed to the other DDUs. Hence, an engineer from the central team was able to spot this quickly as opposed to the one from the brand's DDU.

It would be nice to have a solution where the addition or deletion of email IDs (individual/group/SA) in an ACL would not lead to failures. Or have a single source for email IDs that can be referred by different DDUs in their ACLs - any modifications just have to be done once, centrally.

**Comment by tusharchou** [comment]
Thank you
